src/serac/infrastructure/profiling.cpp

```diff
@@ -20,6 +19,57 @@ std::optional<cali::ConfigManager> mgr;
 }  // namespace
 #endif
 
+void initialize(std::string options, MPI_Comm comm)
+{
+#ifdef SERAC_ENABLE_PROFILING
+  // Initialize Adiak
+  adiak::init(&comm);
+
+  adiak::launchdate();
+  adiak::executable();
+  adiak::cmdline();
+  adiak::clustername();
+  adiak::jobsize();
+  adiak::walltime();
+  adiak::cputime();
+  adiak::systime();
+
+  // Initialize Caliper
+  mgr = cali::ConfigManager();
+  auto check_result = mgr->check(options.c_str());
+
+  if (check_result.empty()) {
+    mgr->add(options.c_str());
+  } else {
+    SLIC_WARNING_ROOT("Caliper options invalid, ignoring: " << check_result);
+  }
+
+  // Defaults, should probably always be enabled
+  mgr->add("event-trace,runtime-report,spot");
+  mgr->start();
+#else
+  // Silence warnings
+  static_cast<void>(options);
```

Comment:
You can use this since we are C++17:

https://en.cppreference.com/w/cpp/language/attributes/maybe_unused